NetSerializer/Primitives.cs

```diff
@@ -255,16 +256,146 @@ public static void ReadPrimitive(Stream stream, out float value)
 		public static void WritePrimitive(Stream stream, double value)
 		{
 			ulong v = (ulong)BitConverter.DoubleToInt64Bits(value);
-			WriteVarint64(stream, v);
+			WriteUInt64(stream, v);
 		}
 
 		public static void ReadPrimitive(Stream stream, out double value)
 		{
-			ulong v = ReadVarint64(stream);
+			ulong v = ReadUInt64(stream);
 			value = BitConverter.Int64BitsToDouble((long)v);
 		}
 #endif
 
+		private static void WriteUInt16(Stream stream, ushort value)
+	    {
+	        stream.WriteByte((byte) value);
+	        stream.WriteByte((byte) (value >> 8));
+	    }
+
+		private static ushort ReadUInt16(Stream stream)
+		{
+            ushort a = 0;
+
+            for (var i = 0; i < 16; i += 8)
+            {
+                var val = stream.ReadByte();
+                if (val == -1)
+                    throw new EndOfStreamException();
+
+                a |= (ushort) (val << i);
+            }
+
+            return a;
+		}
```

Comment:
Shouldn't these two methods be part of the next commit ("Add support for .NET 5 Half type.")?

Reply:
Yeah I was kind of debating that as well but figured this would make the diffs cleaner since now all these uint read/write methods are in the same commit.

Reply:
The reasons why I think it should not be part of the commit:
- This commit is "improve float serialization", i.e. improve what we already have
- The new methods are not used (so they cannot even improve the float serialization)
- The commit message doesn't mention adding new feature
- If you do add a note to the commit message, it would probably be something like this at the end of the message "Also, add xyz", which is a clear hint that the code is probably really not part of the commit.